packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -180,18 +233,45 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     });
   }
 
+  const getSelectedIndexes = () => {
+    const indexArray: number[] = [];
+    isMultipleEnabled(valueProp, multiple) &&
+      valueProp?.forEach((value) => {
+        indexArray.push(
+          optionsRef.current.findIndex((option) => option.value === value),
+        );
+      });
+    return indexArray;
+  };
+
   // Reducer where all the component-wide state is stored
   const [{ isOpen, selectedIndex, focusedIndex }, dispatch] = React.useReducer(
     comboBoxReducer,
     {
       isOpen: false,
-      selectedIndex: valueProp
+      selectedIndex: isMultipleEnabled(valueProp, multiple)
+        ? valueProp
+          ? getSelectedIndexes()
+          : []
+        : valueProp
         ? optionsRef.current.findIndex((option) => option.value === valueProp)
         : -1,
       focusedIndex: -1,
     },
   );
 
+  const selectedOptions = React.useMemo(() => {
+    if (selectedIndex == null) {
+      return undefined;
+    }
+    const item = isMultipleEnabled(selectedIndex, multiple)
+      ? selectedIndex != undefined && selectedIndex.length > 0
+        ? selectedIndex.map((index) => optionsRef.current[index])
```

Comment:
Again hard to understand what is going on here with the nested ternaries. Could you explain/simplify?

Reply:
Added some comments